pkg/handler/handler.go

```diff
@@ -133,80 +150,101 @@ func getContainersToSkip(annotationDomain string, pod *corev1.Pod) map[string]bo
 	return skippedNames
 }
 
-func (m *Modifier) addEnvToContainer(container *corev1.Container, tokenFilePath, roleName string, regionalSTS bool) bool {
+func (m *Modifier) addEnvToContainer(container *corev1.Container, tokenFilePath string, patchConfig *podPatchConfig) bool {
 	var (
-		reservedKeysDefined   bool
-		regionKeyDefined      bool
-		regionalStsKeyDefined bool
+		webIdentityKeysDefined          bool
+		containerCredentialsKeysDefined bool
+		regionKeyDefined                bool
+		regionalStsKeyDefined           bool
 	)
-	reservedKeys := map[string]string{
+	webIdentityKeys := map[string]string{
 		"AWS_ROLE_ARN":                "",
 		"AWS_WEB_IDENTITY_TOKEN_FILE": "",
 	}
+	containerCredentialsKeys := map[string]string{
+		pkg.AwsEnvVarContainerCredentialsFullUri:     "",
+		pkg.AwsEnvVarContainerAuthorizationTokenFile: "",
+	}
 	awsRegionKeys := map[string]string{
 		"AWS_REGION":         "",
 		"AWS_DEFAULT_REGION": "",
 	}
 	stsKey := "AWS_STS_REGIONAL_ENDPOINTS"
 	for _, env := range container.Env {
-		if _, ok := reservedKeys[env.Name]; ok {
-			reservedKeysDefined = true
+		if _, ok := webIdentityKeys[env.Name]; ok {
+			klog.V(4).Infof("Web identity env variable %s is already defined in the pod spec", env)
+			webIdentityKeysDefined = true
+		}
+		if _, ok := containerCredentialsKeys[env.Name]; ok {
```

Comment:
What happens if only one of these env vars are set by the customer?  Does that result in a working pod state?  Right now, if only 1 is set, we say `containerCredentialsKeysDefined = true` and we skip patching entirely.  Wondering if it makes more sense to check for both and only skip patching if they are both set.  Might be confusing to diverge from the web identity logic here but trying to understand what UX makes the most sense.  Also not sure what the patch behavior is when only 1 is set (does the set env var get overwritten?).

Reply:
Had a conversation off-line and am satisfied to keep this consistent with web identity. We can consider alternative behavior in the future gated by a flag or something if the use case arises